pkg/cmd/pulumi/stack_init.go

```diff
@@ -74,119 +69,145 @@ func newStackInitCmd() *cobra.Command {
-						"To create a stack in an organization, " +
-						"use the format <org-name>/<stack-name> (e.g. `acmecorp/dev`).\n")
-				}
-
-				name, nameErr := promptForValue(false, "stack name", "dev", false, b.ValidateStackName, opts)
-				if nameErr != nil {
-					return nameErr
-				}
-				stackName = name
-			}
-
-			if stackName == "" {
-				return errors.New("missing stack name")
-			}
-
-			if err := b.ValidateStackName(stackName); err != nil {
-				return err
-			}
-
-			stackRef, err := b.ParseStackReference(stackName)
-			if err != nil {
-				return err
-			}
-
-			proj, root, projectErr := readProject()
-			if projectErr != nil && !errors.Is(projectErr, workspace.ErrProjectNotFound) {
-				return projectErr
-			}
-
-			// Backend-specific config options. Currently only applicable to the HTTP backend.
-			createOpts, err := validateCreateStackOpts(stackName, b, teams)
-			if err != nil {
-				return err
-			}
-
-			newStack, err := createStack(ctx, b, stackRef, root, createOpts, !noSelect, secretsProvider)
-			if err != nil {
-				return err
-			}
-
-			if stackToCopy != "" {
-				if projectErr != nil {
-					return projectErr
-				}
-
-				// load the old stack and its project
-				copyStack, err := requireStack(ctx, stackToCopy, stackLoadOnly, opts)
-				if err != nil {
-					return err
-				}
-				copyProjectStack, err := loadProjectStack(proj, copyStack)
-				if err != nil {
-					return err
-				}
-
-				// get the project for the newly created stack
-				newProjectStack, err := loadProjectStack(proj, newStack)
-				if err != nil {
-					return err
-				}
-
-				// copy the config from the old to the new
-				return copyEntireConfigMap(copyStack, copyProjectStack, newStack, newProjectStack)
-			}
-
-			return nil
+			return sicmd.Run(ctx, args)
 		}),
 	}
 	cmd.PersistentFlags().StringVarP(
-		&stackName, "stack", "s", "", "The name of the stack to create")
+		&sicmd.stackName, "stack", "s", "", "The name of the stack to create")
 	cmd.PersistentFlags().StringVar(
-		&secretsProvider, "secrets-provider", "default", possibleSecretsProviderChoices)
+		&sicmd.secretsProvider, "secrets-provider", "", possibleSecretsProviderChoices)
 	cmd.PersistentFlags().StringVar(
-		&stackToCopy, "copy-config-from", "", "The name of the stack to copy existing config from")
+		&sicmd.stackToCopy, "copy-config-from", "", "The name of the stack to copy existing config from")
 	cmd.PersistentFlags().BoolVar(
-		&noSelect, "no-select", false, "Do not select the stack")
-	cmd.PersistentFlags().StringArrayVar(&teams, "teams", nil, "A list of team "+
+		&sicmd.noSelect, "no-select", false, "Do not select the stack")
+	cmd.PersistentFlags().StringArrayVar(&sicmd.teams, "teams", nil, "A list of team "+
 		"names that should have permission to read and update this stack,"+
 		" once created")
 	return cmd
 }
 
+// stackInitCmd implements the `pulumi stack init` command.
+type stackInitCmd struct {
+	// Flags:
```

Comment:
Should there be more to this comment? We haven't tagged the flags explicitly like this in the other refactors.

Reply:
This was mostly for helping inform the grouping. I agree this may not be necessary. Will delete.